theater/start_generator.py

```diff
@@ -260,22 +442,31 @@ def generate_airbase_defense_group(airbase_defense_group_id, ground_obj:TheaterG
         ground_obj.groups.append(group)
 
 
-def find_location(on_ground, near, theater, min, max, others, is_base_defense=False) -> typing.Optional[Point]:
+# TODO: https://stackoverflow.com/a/19482012/632035
+# A lot of the time spent on mission generation is spent in this function since
+# just randomly guess up to 1800 times and often fail. This is particularly
+# problematic while trying to find placement for navies in Nevada.
```

Comment:
Yes, that's the main reason why campaign generation takes so long sometimes. I'm working on presets locations in a set of miz files, so we don't have to guess randomly. 
I haven't commited anything yet. But this should also solve the problems with SAM site being located in the middle of forests and such. 😄